gui/gui-manager.cpp

```diff
@@ -329,20 +333,14 @@ void GuiManager::redraw() {
 
 			if (_displayTopDialogOnly) {
 				// When displaying only the top dialog clear the screen
-				if (_redrawStatus == kRedrawOpenDialog) {
```

Comment:
I am not sure I follow you on this reading.
`clearAll` clears the overlay before grabbing it. This is done like this to fetch the game screen underneath when the platform doesn't support layers and transparency (`clearOverlay` copies the game screen on the overlay, `grabOverlay` lets the GUI do draw operations on it).
It's called on full redraw, dialog close and dialog open when only top is drawn.

In `open` case, the previous layers are not redrawn, only the new one is drawn above the previous ones.

`close` is really like a full redraw, nothing else must be thought about it. Our stack supports efficient pushes but requires full redraw when popping.

I don't see where the fillScreen(0) is used in this code.
I don't understand the relation between `GuiManager::redraw` and `OptionsDialog::testGraphicsSettings`. Please don't design `redraw` around a specific case.

I propose a rewrite of `GuiManager::redraw` in https://github.com/lephilousophe/scummvm/commit/rework-redraw where I split the only top dialog from the other code.
I think it's more readable in this branch with minimal code duplication.

Reply:
Yeah, sorry, I should have been more specific. What I wanted to say is that `_displayTopDialogOnly` is set only via `GuiManager::displayTopDialogOnly` which is called only from one place - `OptionsDialog::testGraphicsSettings` in a very unusual fashion which implied to me that it could have been perhaps reworked in a way that this flag wouldn't be needed in the first place.

However since you nicely separated this case out of `redraw`, I have no problem with keeping the code as is. I was just worried about all those `clearAll` calls, which are super expensive on my platform (but as mentioned, this code would be never called there so it was just a theoretical question).